### PR TITLE
fix: error if eval()-ing repr(SchemaField)

### LINKED ISSUE #1041: SchemaField cannot initialize from __repr__
#### Environment details

  - OS type and version: macOS Big Sur 11.6.1
  - Python version: `3.9.7`
  - pip version: `21.3.1`
  - `google-cloud-bigquery` version: `2.29.0`

#### Description

Prior versions of `google-cloud-bigquery` (`2.3.1`) allowed `SchemaField` instantiation based on an existing `SchemaField` `__repr__`.

#### Use Case

1. System generates list of `SchemaField` and passes it through a Jinja template as part of building hundreds of Airflow DAGs.
2. Jinja prints `__repr__` from the `SchemaField` into the templated `.py`. 
3. Airflow picks up the `.py` files and executes them. 
4. Execution results in the below error.

#### Code example

In version `2.3.1`

```python
from google.cloud.bigquery import SchemaField
from google.cloud import bigquery
bigquery.__version__
'2.3.1'

# This works
field = SchemaField('name', 'STRING', 'REQUIRED', 'Name') 
field.__repr__()
"SchemaField('name', 'STRING', 'REQUIRED', 'Name', (), ())"
# Jinja prints ^^ to a .py file
# When the file is executed the SchemaField is created without issue
```

Version `2.29.0`:

```python
from google.cloud.bigquery import SchemaField
from google.cloud import bigquery
bigquery.__version__
'2.29.0'

# This raises an error for the tuple in the last field
field = SchemaField('name', 'STRING', 'REQUIRED', 'Name') 
field.__repr__()
"SchemaField('name', 'STRING', 'REQUIRED', 'Name', (), ())"
# Jinja prints ^^ to a .py file
# When the file is executed, an error is raised:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/wes.roach/.pyenv/versions/upgr-dbt/lib/python3.9/site-packages/google/cloud/bigquery/schema.py", line 125, in __init__
    policy_tags.to_api_repr() if policy_tags is not None else None
AttributeError: 'tuple' object has no attribute 'to_api_repr'
```

Setting the `policy_tags` to `None` still results in `()`.

```python
field = SchemaField('name', 'STRING', 'REQUIRED', 'Name', (), None)
field.__repr__()
"SchemaField('name', 'STRING', 'REQUIRED', 'Name', (), ())"
```

### PR DESCRIPTION
**Pull request by plamut**
Fixes #1041.

If policy tags are not set, the corresponding key should be `None` to denote a missing value. An empty tuple should only be used when the schema has explicitly empty `PolicyTagList` assigned.

**PR checklist:**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: tests/unit/test_schema.py [hunk 1]
```diff
@@ -510,9 +510,18 @@ def test___hash__not_equals(self):
...
 
+    def test___repr__evaluable(self):
+        field = self._make_one("field1", "STRING", "REQUIRED", "Description")
+        field_repr = repr(field)
+        SchemaField = self._get_target_class()  # needed for eval  # noqa
+
+        evaled_field = eval(field_repr)
+
+        assert field == evaled_field
+
```
Comment (tseaver) [inline tests/unit/test_schema.py:524]: For completeness, do we need a round-trip test for the case where there is a policy tuple?

Comment (plamut) [reply tests/unit/test_schema.py:524]: Would not hurt, I'll add it.

**Review by tseaver — APPROVED** [review]

File: tests/unit/test_schema.py (hunk 1 above)
Comment (plamut) [reply tests/unit/test_schema.py:524]: Guess what, that extra test actually revealed deeper issues in the `repr()` of `SchemaField` and `PolicyTagList`.

**Review by tswast — APPROVED** [review]
Thanks!

File: tests/unit/test_schema.py (hunk 1 above)
Comment (tseaver) [reply tests/unit/test_schema.py:524]: W00t!

**Pull request merged**